docs/api/List.md

```diff
@@ -1 +1,156 @@
-# TODO
+# List Module
+***Note that while these are separated into categories here based on behavior, they are all called from _.List***
+
+## Delegation
+### IList\<T\> Resolve\<T\>(IList<T> list)
+Executes the list of functions and returns the results
+``` csharp
+
+Func<int>[] myArray = { ()=>1, ()=>2, ()=>3, ()=>4 };
+var myArraySnapshot = _.List.Resolve(myArray);
+
+int[] snapshotResult = myArraySnapshot(); // returns { 1, 2, 3, 4 }
+
+```
+# Manipulate
+### bool Swap\<T\>(IList\<T\> collection, int i,int j)
+Swaps two items in a list, returns true if it was successful
+``` csharp
+var ls = new[]{1,2,3,4};
+_.List.Swap(ls, 0, 1) // returns true, ls is now {2,1,3,4}
+```
+
+### IList\<T\> Shuffle\<T\>(IList\<T\> ls)
+Returns a list of randomly shuffle items from passed list 
+``` csharp
+_.List.Shuffle(new int[] {1, 2, 3}) // could return {3,2,1} or {2,3,1} or {1,3,2} 
+```
+
+
+### IList\<T\> Shuffle\<T\>(IList\<T\> ls, bool inplace)
+Returns a list of randomly shuffle items from passed list, if inplace is passed as true shuffling is done in same list and that list is returned
+``` csharp
+var ls = new int[] {1, 2, 3};
+var nls = _.List.Shuffle(ls,true) // could return {3,2,1} or {2,3,1} or {1,3,2}
+ls == nls // this will be true
+```
+
+### void Rotate\<T\>(IList\<T\> list, int change);
+performs a right rotate on a list
```

Comment:
should include details in the comments, I (and many other people programming in .NET) aren't going to know random ruby utility syntax